relay_hub/storage.py

```
@@ -0,0 +1,622 @@
+                
+                # Atomically claim the selected messages
+                placeholders = ','.join('?' for _ in message_ids)
+                cur.execute(
+                    f"""
+                    UPDATE messages
+                       SET status = ?,
+                           claim_owner = ?,
+                           claim_expires_at = ?,
+                           updated_at = ?
+                     WHERE id IN ({placeholders})
+                    """,
```

Ruff (0.13.1)

Possible SQL injection vector through string-based query construction

(S608)